client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -214,18 +218,31 @@ export class DisplayTable extends React.Component {
       .thru((csv_data) => d3.csvFormatRows(csv_data))
       .value();
 
+    const showing_search = _.isUndefined(show_search)
+      ? _.size(data) > 6
+      : show_search;
+    const showing_sort = _.isUndefined(show_sort)
+      ? _.size(data) > 2
+      : show_sort;
+    const showing_column_select = _.isUndefined(show_column_select)
+      ? _.size(all_ordered_col_keys) > 2
+      : show_column_select;
+    const showing_export_icons = _.isUndefined(show_export_icons)
+      ? _.size(all_ordered_col_keys) * _.size(data) > 15
+      : show_export_icons;
+
     const util_components_default = {
-      copyCsvUtil: (
+      copyCsvUtil: showing_export_icons && (
         <DisplayTableCopyCsv key="copyCsvUtil" csv_string={csv_string} />
       ),
-      downloadCsvUtil: (
+      downloadCsvUtil: showing_export_icons && (
         <DisplayTableDownloadCsv
           key="downloadCsvUtil"
           csv_string={csv_string}
           table_name={table_name}
         />
       ),
-      columnToggleUtil: (
+      columnToggleUtil: showing_column_select && (
```

Comment:
So `_.size(all_ordered_col_keys) > 2` straight in here. But also, we'd want to check that `initial_visible` is true for all columns for this as well to make sure there are no hidden columns